### PR TITLE
Move duplicate getCtfTrackRef function to its own file

### PR DESCRIPTION
**Pull request by guitargeek**
Just some reorganization of duplicate code in RecoEgamma, which also goes in the direction of slimming down a bit GsfElectronAlgo aka. "the electron god class".

Are there any convention related to names of files holding free functions like this? And do you maybe share my sentiment that the function should be maybe given a more clear name like `getClosestCtfToGsf`?

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by guitargeek** [comment]
Hi @perrotta, with https://github.com/cms-sw/cmssw/pull/25977 being merged soon I will have to do a rebase here. Do you have any suggestion about e.g. the file name so I can do the rebase and implement comments in one go? Thanks!

File: RecoEgamma/EgammaElectronAlgos/interface/getCtfTrackRef.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef getCtfTrackRef_h
```
Comment (perrotta) [inline RecoEgamma/EgammaElectronAlgos/interface/getCtfTrackRef.h]: Include guard should include the full path

**Review by perrotta — COMMENTED** [review]
About the name of the file: you could do something similar as, e.g., EgAmbiguityTools in the same package: GsfElectronTools (?) with a namespace which includes the functions in it (only this one, by now).
About the name of the function, since it is only addressed in a couple of places in cmssw, I think you can rename it in some more appropriate way if you think so.

File: RecoEgamma/EgammaElectronAlgos/interface/getCtfTrackRef.h (hunk 1 above)
Comment (guitargeek) [reply RecoEgamma/EgammaElectronAlgos/interface/getCtfTrackRef.h]: Ok! Thanks for giving your opinion.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test
(hoping that the issue described in #26002 can be solved now)

**Comment by perrotta** [comment]
+1
- It just moves a utility function into a dedicated file, so that it can be called from the different classes without code duplication
- No change expected in outputs: jenkins tests pass and show no changes

**Comment by fabiocos** [comment]
+1

**Pull request merged**